### ISSUE TITLE
Undefined variables in list comprehensions after latest update

### ISSUE THREAD

**Issue by HenrikOssipoff** [Microsoft/python-language-server#830]
After the latest update to the Python extension that dropped today we're seeing a bunch of undefined variables showing up in list comprehensions. I'm assuming it's to do with some update in the language server. This snippet of code demonstrates it:

```python
def func_a(value):
    return value


def func_b():
    list_a = [1, 2, 3, 4]

    return [func_a(value=elem) for elem in list_a]
```

In the above code, `elem` in the last line will be undefined inside the function call to `func_a`.

Version of the extension used: 2019.3.6139.

**Comment by CaselIT** [comment]
Came across this issue as well. It only happens when returning the list direclty
![image](https://user-images.githubusercontent.com/16175304/55250381-55af7680-524e-11e9-95d1-2a522a604c17.png)

When assigning it to a variable it works correctly
![image](https://user-images.githubusercontent.com/16175304/55250414-665fec80-524e-11e9-8d08-331d142177ed.png)

**Comment by jakebailey** [comment]
This is fixed with #819.
